File: app/src/main/java/fr/openium/examplepam/model/Call.java
package fr.openium.examplepam.model;

import java.util.Date;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Call {
    @PrimaryKey(autoGenerate = true)
    public int uid;

    @ColumnInfo
    public String contactName;

    @ColumnInfo
    public int length;

    @ColumnInfo
    public Date startDate;

    public Call() {

    }

    public Call(String contactName, long startDate) {
        this.contactName = contactName;
        this.startDate = new Date(startDate);
    }
}

File: app/src/main/java/fr/openium/examplepam/model/Call.kt
package fr.openium.examplepam.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.*

@Entity
data class Call(
    @PrimaryKey(autoGenerate = true)
    var uid: Long? = null,
    @ColumnInfo
    var contactName: String? = null,
    @ColumnInfo
    var length: Long = 0,
    @ColumnInfo
    var startDate: Date = Date()
)